// Copyright (c) YugaByte, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
// in compliance with the License.  You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software distributed under the License
// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
// or implied.  See the License for the specific language governing permissions and limitations
// under the License.
//

syntax = "proto2";

package yb.tserver;

option java_package = "org.yb.tserver";

import "yb/tserver/tserver.proto";

service TabletServerBackupService {
  rpc TabletSnapshotOp(TabletSnapshotOpRequestPB)
      returns (TabletSnapshotOpResponsePB);
}

message TabletSnapshotOpRequestPB {
  enum Operation {
    UNKNOWN = 0;
    CREATE = 1;
    RESTORE = 2;
    DELETE = 3;
  }

  // UUID of server this request is addressed to.
  optional bytes dest_uuid = 1;

  optional Operation operation = 2;

  optional bytes snapshot_id = 3;

  // Tablet to snapshot.
  optional bytes tablet_id = 4;

  optional fixed64 propagated_hybrid_time = 5;

  // Override the snapshot directory. Used in restoring sys catalog to avoid running initdb.
  optional string snapshot_dir_override = 6;
}

message TabletSnapshotOpResponsePB {
  optional TabletServerErrorPB error = 1;

  optional fixed64 propagated_hybrid_time = 2;
}
